### ISSUE TITLE
Add inline_return_spec to enforce breaking on spec and callback

### ISSUE THREAD

**Issue by rjcoelho** [AdRoll/rebar3_format#229]
From 
```
-spec func(A :: type()) -> Return :: type().
```

I want to enforce break on return no matter what paper value I have
```
-spec func(A :: type()) -> 
    Return :: type().
```

**Comment by elbrujohalcon** [comment]
Good idea. I'll put it in milestone `0.10.0`. If you want to give it a try and send a PR, that would also be lovely.

**Comment by rjcoelho** [comment]
@elbrujohalcon how do I distinguish the return type from the list of function argument types ?
i'm here

**Comment by elbrujohalcon** [comment]
@rjcoelho each element within `Types` contains both the arguments and the result… because a spec can have multiple clauses, like…

```erlang
-spec something(With, Multiple, Clauses) -> Result
                   when With :: with,
                        Multiple :: multiple,
                        Clauses :: clauses,
                        Result :: result;
               (With, Multiple, Clauses) -> Result
                   when With :: [with],
                        Multiple :: [multiple],
                        Clauses :: [clauses],
                        Result :: [result];
               ({With :: with}, {Multiple :: multiple}, {Clauses :: clauses}) -> {Result :: result}.
```

You have to go deeper. The code you're looking for is in either `lay_clauses(Types, spec, Ctxt1)` or `lay_simple_spec/3`.

**Comment by elbrujohalcon** [comment]
Even deeper… I think what you're looking for is how to format a `function_type`…
Somewhere around here:
```erlang
        function_type ->
            {Before, After} =
                case Ctxt#ctxt.clause of
                    spec ->
                        {"", ""};
                    _ ->
                        {"fun(", ")"}
                end,
            Ctxt1 = (reset_prec(Ctxt))#ctxt{clause = undefined},
            D1 = case erl_syntax:function_type_arguments(Node) of
                     any_arity ->
                         text("(...)");
                     Arguments ->
                         As = lay_items(Arguments, Ctxt1, fun lay/2),
                         beside(text("("), beside(As, lay_text_float(")")))
                 end,
            D2 = lay(erl_syntax:function_type_return(Node), Ctxt1),
            beside(lay_text_float(Before),
                   sep([beside(D1, lay_text_float(" ->")),
                        nest(Ctxt#ctxt.break_indent, beside(D2, lay_text_float(After)))]));
```

**Comment by elbrujohalcon** [comment]
Oh, and you have to check how to format a `constrained_function_type` as well :)

**Comment by rjcoelho** [comment]
@elbrujohalcon 0.10 CHANGELOG says `inline_return_spec` is  implemented but i cannot  find it

**Comment by elbrujohalcon** [comment]
@rjcoelho The changelog is wrong… that's because `github_changelog_generator` sees this ticket as _Closed_ and it thinks it's solved. 🤦‍♂️ 
Feel free to reopen the ticket if you still want it and, of course, a PR will still be appreciated.

**Comment by elbrujohalcon** [comment]
I fixed the CHANGELOG with ee5beed, by adding the _wontfix_ label to the ticket.
If you reopen this, remind me to remove the label please.
